packages/flutter_tools/templates/app/lib/main.dart.tmpl

```diff
@@ -38,111 +38,182 @@ void main() {
+            // set in the AnimatedContainer, above.
+            pointRounding: 0.4,
+            valleyRounding: 0.4,
+            innerRadiusRatio: .5,
+          ),
+        ),
         child: Column(
-          // Column is also a layout widget. It takes a list of children and
-          // arranges them vertically. By default, it sizes itself to fit its
-          // children horizontally, and tries to be as tall as its parent.
-          //
-          // Invoke "debug painting" (press "p" in the console, choose the
-          // "Toggle Debug Paint" action from the Flutter Inspector in Android
-          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
-          // to see the wireframe for each widget.
+          // Column is a layout widget. It takes a list of children and arranges
+          // them vertically. By default, it sizes itself to fit its children
+          // horizontally, and tries to be as tall as its parent.
           //
           // Column has various properties to control how it sizes itself and
           // how it positions its children. Here we use mainAxisAlignment to
           // center the children vertically; the main axis here is the vertical
           // axis because Columns are vertical (the cross axis would be
           // horizontal).
+          //
+          // TRY THIS: Invoke "debug painting" (press "p" in the console, or
+          // choose the "Toggle Debug Paint" action in the IDE), to see the
+          // wireframe for each widget.
           mainAxisAlignment: MainAxisAlignment.center,
           children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
-            ),
             Text(
               '$_counter',
-              style: Theme.of(context).textTheme.headlineMedium,
+              style: Theme.of(context).textTheme.displayMedium,
             ),
           ],
         ),
       ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
-        child: const Icon(Icons.add),
+      floatingActionButton: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Padding(
+            padding: const EdgeInsetsDirectional.only(end: 8.0),
+            child: FloatingActionButton(
+              // Setting the onPressed callback to null when the counter is at
+              // its minimum value will disable to button so that the counter
+              // can't go below its minimum.
+              onPressed: _counter > 3 ? () => _incrementCounterBy(-1) : null,
+              tooltip: 'Decrement',
+              disabledElevation: 0,
+              child: const Icon(Icons.remove),
+            ),
+          ),
+          FloatingActionButton(
+            onPressed: () => _incrementCounterBy(1),
+            tooltip: 'Increment',
+            child: const Icon(Icons.add),
+          ),
```

Comment:
isn't having two FABs explicitly against the material guidelines?

Reply:
I don't see any mention of that here, but maybe I missed it: https://m3.material.io/components/floating-action-button/overview